### ISSUE TITLE
Force unwrapping false positive with Bool(s)

### ISSUE THREAD

**Issue by dral3x** [realm/SwiftLint#1262]
In this function, the exclamation mark in the `isNetworkReachable` definition is recognized as a force unwrap, which clearly is not.

```
func calculateNetworkState(_ flags: SCNetworkReachabilityFlags) -> NetworkManagerState {
        
        let isReachable = flags.contains(.reachable)
        let needsConnection = flags.contains(.connectionRequired)
        let canConnectionAutomatically = flags.contains(.connectionOnDemand) || flags.contains(.connectionOnTraffic)
        let canConnectWithoutUserInteraction = canConnectionAutomatically && !flags.contains(.interventionRequired)
        let isNetworkReachable = isReachable && (!needsConnection || canConnectWithoutUserInteraction)
        
        if !isNetworkReachable {
            return .notReachable
        }
        
        // On simulator: if the network is reachable it can be reachable only via WiFi
        if TARGET_OS_SIMULATOR != 0 {
            return .reachableViaWiFi
        }
        
        return flags.contains(.isWWAN) ? .reachableViaCellular : .reachableViaWiFi
    }
```

I'm using swiftlint version 0.16.1 (via gem install).

**Comment by marcelofabri** [comment]
@dral3x This should be fixed on `master` (#1215). Could you try linting with it?

**Comment by dral3x** [comment]
@marcelofabri any guide or hint on how to build swiftlint from source?
Thanks

**Comment by marcelofabri** [comment]
https://github.com/realm/SwiftLint#compiling-from-source should cover it. I'd recommend to uninstall the homebrew version first to be sure that you're using the built version.

**Comment by dral3x** [comment]
Sorry for the previous question, I didn't found that information before asking but I did few seconds later.
I will let you know

**Comment by marcelofabri** [comment]
No worries! BTW, I've tested with `0.16.1` and `master` and the violation was only triggered on `0.16.1`.

**Comment by dral3x** [comment]
Just checked and yes, with `master` build the violation is not triggered anymore.
Any ETA on next public release (via brew) ?

Thank you!

**Comment by marcelofabri** [comment]
@dral3x We have several PRs open right now, so my guess is that we'll want to merge some of them before cutting a new release.
